src/agent/configuration_parser/tests/configuration_parser_test.cpp

```diff
@@ -2,11 +2,81 @@
 
 #include <gtest/gtest.h>
 
+#include <filesystem>
+#include <fstream>
 #include <map>
 #include <memory>
 
 using namespace configuration;
 
+class ConfigurationParserFileTest : public ::testing::Test
+{
+protected:
+    std::filesystem::path tempConfigFilePath;
+
+    void SetUp() override
+    {
+        tempConfigFilePath = "temp_wazuh-agent.yml";
+
+        std::ofstream outFile(tempConfigFilePath);
+        outFile << R"(
+            agent:
+                server_url: https://myserver:28000
+                registration_url: https://myserver:56000
+            inventory:
+                enabled: false
+                interval: 7200
+                scan_on_start: false
+            logcollector:
+                enabled: false
+                localfiles:
+                - /var/log/other.log
+                reload_interval: 120
+                file_wait: 1000
+        )";
+        outFile.close();
+    }
+
+    void TearDown() override
+    {
+        std::filesystem::remove(tempConfigFilePath);
```

Comment:
How necessary is this remove? I ask before if this crashes somehow the file won't be cleaned up.

Reply:
If the failure is due to an `EXPECT_*` the TearDown would be executed, the `EXPECT_*` allows the test to continue to the end, registering any failure, but without interrupting the execution. That is why I decided to use `EXPECT_*` instead of `ASSERT_*`.
However, I could add some more controls in case there is an exception for some other reason.
In the tests:
```
    try
    {
        const auto parser = std::make_unique<configuration::ConfigurationParser>(m_tempConfigFilePath);
    }
    catch(const std::exception& e)
    {
        std::filesystem::remove(m_tempConfigFilePath);
        throw;
    }
```
What do you think?

Reply:
Right, the issue would come up if the application crashes before executing the `TearDown`. The problem with this try catch is that it would remove the file, but wouldn't create the parser. I think that checking that there's no file, and removing them if there are any, in the constructor, would be enough.

Reply:
by default, ofstream overwrites the file in case it exists.